### ISSUE TITLE
N Outcomes - proptestn

### ISSUE THREAD

**Issue by PeterC-alfaisal** [silkyproject/silkyR#418]
There seems to be a bug in the Frequencies > N Outcomes analysis item.
It shows Expected Proportions for levels that have been filtered out. Since a filter has removed the items from analysis, they should not appear in the dropdown for Expected Proportions.

**Comment by jonathon-love** [comment]
Yeah, this is a known issue, and I don't think we've got a good solution for it. The options UI is unaware of whether all of a particular level has been filtered out. This is a limitation of our architecture for the time being. Not an easy fix.

**Comment by PeterC-alfaisal** [comment]
OK, thanks for the super quick response.
As you probably also know, the calculation of expected proportions is also affected unless a ratio of 0 is entered for the filtered items. The chi square statistic calculation is unavoidably affected.

The issue affects the N Outcomes in jeva so that it actually crashes when I apply a filter, giving:
Error in private$.run(): dims [product 2] do not match the length of object [4]
private$.run()

**Comment by jonathon-love** [comment]
i think this might actually be fixed in a recent version

**Comment by PeterC-alfaisal** [comment]
Yes, is fixed in version 2.7.37, but flags same error in jeva
